src/js/index.js

```diff
@@ -46,44 +47,58 @@ function handleDifficulty(difficultyChanged) {
 	return false;
 }
 
+function handleSort() {
+	Array.from(contentEl.children)
+		.map(child => contentEl.removeChild(child))
+		.sort(sort[sortingInput.value])
+		.forEach(sortedChild => contentEl.append(sortedChild));
+}
+
 function handleTags() {
 	const tags = selectr.getValue();
 
-	if (tags.length === 0) {
-		history.pushState(null, '', window.location.pathname);
-		return;
-	}
-
 	Array.from(contentEl.querySelectorAll('.item')).forEach(el => {
-		const show = tags.reduce((sho, tag) => sho || el.classList.contains(`tag-${tag}`), false);
+		const show = (showExpired.checked || !el.classList.contains('tag-expired')) &&
+			tags.reduce((sho, tag) => sho || el.classList.contains(`tag-${tag}`), tags.length === 0);
 		if (!show) {
 			el.classList.remove('visible');
 		}
 	});
 
-	if (!search) {
-		return;
-	}
-
 	search.set('tags', tags.join(' '));
-	const newRelativePathQuery = `${window.location.pathname}?${search.toString()}`;
-	history.pushState(null, '', newRelativePathQuery);
+
+	search.set('expired', showExpired.checked || '');
 }
 
-function handleSort() {
-	Array.from(contentEl.children)
-		.map(child => contentEl.removeChild(child))
-		.sort(sort[sortingInput.value])
-		.forEach(sortedChild => contentEl.append(sortedChild));
+function updateUrl() {
+	let nextPath = window.location.pathname;
+
+	const emptyParams = [];
```

Comment:
We can expand the variable names, but 👍

Reply:
> This logic can be replaced by:

It can be, but doing so would make the code not very readable and maintainable (in my opinion, at least), and that's more important than having it more optimized as a one-liner. I think we should stick with the current `for...of` loop.

Reply:
WDYT of:

```javascript
Array.from(search).forEach(([key, value]) => {
	if (!value.trim()) {
		search.delete(key);
	}
});
```

It cut the needs for a intermediate array, I think it keeps code dry. If it still makes code less readable for you, we'd go your way.

Reply:
Either way is good with me 🎉 

Just putting this out there; this won't have an effect on performance by any means because we're dealing with literally 5 elements at most - it's just fun to think about!

> It cut the needs for a intermediate array

That's completely true, but it requires 2 iterations over the data because `Array.from()` does something similar to the `for...of` internally, and then we iterate over the returned array

Reply:
> It cut the needs for a intermediate array, I think it keeps code dry. If it still makes code less readable for you, we'd go your way.

@aslafy-z I think we should keep the current implementation 🙃

> this won't have an effect on performance by any means because we're dealing with literally 5 elements at most

Also true 😅